sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/ProactiveOpenConnectionsProcessor.java

```diff
@@ -100,7 +100,7 @@ private void submitOpenConnectionTaskOutsideLoopInternal(OpenConnectionTask open
         // to open connection / channels when channels are being closed
         // and RntbdTransportClient / RntbdEndpointProvider are also closed
         // this prevents netty executor classes from entering into IllegalStateException
-        if (this.isClosed.get()) {
+        if (this.endpointProvider.isClosed() || this.isClosed.get()) {
```

Comment:
we should not have this.endpointProvider.isClosed() check

the endpoint can be closed due to two reasons:
1. IdleEndpointTimeout - we need to open connections for this case
2. The transportClient is closed - we will fail the open connection task. And this.isClosed.get() is enough